src/main/java/org/rutebanken/tiamat/auth/DefaultAuthorizationService.java

```diff
@@ -67,5 +89,152 @@ public <T extends EntityStructure> Set<String> getRelevantRolesForEntity(T entit
+            return Set.of(ENTITY_CLASSIFIER_ALL_ATTRIBUTES);
+        }
+        return getStopTypesOrSubmode(SUBMODE, false, entity);
+    }
+
+    @Override
+    public Set<String> getLocationBannedSubmodes(boolean canEdit, Point point) {
+        return getLocationStopTypesOrSubmode(canEdit,SUBMODE, false, point);
+    }
+
+    @Override
+    public boolean isGuest() {
+        if (hasNoAuthentications()) {
+            return true;
+        }
+        return roleAssignmentExtractor.getRoleAssignmentsForUser().isEmpty();
+    }
+
+    private Set<String> getStopTypesOrSubmode(String type, boolean isAllowed, EntityStructure entity) {
+        if (hasNoAuthentications()) {
+            return Set.of();
+        }
+        return roleAssignmentExtractor.getRoleAssignmentsForUser().stream()
+                .filter(roleAssignment -> canEditDeleteEntity(entity,roleAssignment.getRole()))
+                .filter(roleAssignment -> roleAssignment.getEntityClassifications() != null)
+                .filter(roleAssignment -> topographicPlaceChecker.entityMatchesAdministrativeZone(roleAssignment, entity))
+                .filter(roleAssignment -> roleAssignment.getEntityClassifications().get(type) != null)
+                .map(roleAssignment -> roleAssignment.getEntityClassifications().get(type))
+                .flatMap(List::stream)
+                .filter(types -> isAllowed != types.startsWith("!"))
+                .map(types -> isAllowed ? types : types.substring(1))
+                .collect(Collectors.toSet());
+    }
+
+    private Set<String> getLocationStopTypesOrSubmode(boolean canEdit, String type, boolean isAllowed, Point point) {
+        if (hasNoAuthentications()) {
+                return Set.of();
+        }
+        if (!canEdit && !isAllowed) {
+            return Set.of(ENTITY_CLASSIFIER_ALL_ATTRIBUTES);
+        }
+        Set<String> stopTypesSubmodes = roleAssignmentExtractor.getRoleAssignmentsForUser().stream()
+                .filter(roleAssignment -> roleAssignment.getEntityClassifications() != null)
+                .filter(roleAssignment -> topographicPlaceChecker.entityMatchesAdministrativeZone(roleAssignment,point ))
+                .filter(roleAssignment -> roleAssignment.getEntityClassifications().get(type) != null)
+                .map(roleAssignment -> roleAssignment.getEntityClassifications().get(type))
+                .flatMap(List::stream)
+                .filter(types -> isAllowed != types.startsWith("!"))
+                .map(types -> isAllowed ? types : types.substring(1))
+                .collect(Collectors.toSet());
+        if (canEdit && stopTypesSubmodes.isEmpty() && isAllowed) {
+            stopTypesSubmodes.add(ENTITY_CLASSIFIER_ALL_ATTRIBUTES);
+        }
+        return stopTypesSubmodes;
+    }
+
+    private boolean filterByRole(RoleAssignment roleAssignment,Object entity) {
```

Comment:
This method is not used by anything.

Reply:
Also ` <T extends EntityStructure> Set<String> getRelevantRolesForEntity(T entity);` this should not be in this interface?

Reply:
These issues are resolved